server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1632,4 +1634,14 @@ static void validateTranslogDurabilitySettings(Settings requestSettings, Cluster
         }
 
     }
+
+    public static void validateCompositeFS(Settings indexSettings) {
+        if (indexSettings.get(IndexModule.INDEX_STORE_TYPE_SETTING.getKey(), "")
+            .equalsIgnoreCase(IndexModule.Type.COMPOSITEFS.getSettingsKey())
+            && !FeatureFlags.isEnabled(FeatureFlags.WRITEABLE_REMOTE_INDEX_SETTING)) {
+            throw new IllegalArgumentException(
+                "ERROR - Composite FS store type can be enabled only if Feature Flag for Writable Remote Index is true"
+            );
+        }
+    }
```

Comment:
What is the plan for migration of existing remote store based directories?

Reply:
Correct @sohami have added a setting `index.store.locality` indicating whether full data (hot index) or partial data (warm index) is cached locally.
For migration, we will add support in the CompositeDirectory itself to support clearing out entries from FileCache, downloading all files to local etc based on migration direction.